docs/models/blockstates/introduction.md

```diff
@@ -0,0 +1,46 @@
+Introduction to Blockstate JSONs
+================================
+
+Blockstate JSONs are Minecraft's way to map "variant strings" to models. A variant string can be absolutely anything, from "inventory" to "power=5" to "I am your father." When the game searches for a model corresponding to a block in the world, it takes the [blockstate][] for that position, and then it uses an `IStateMapper` to find the corresponding blockstate JSON and the variant string within it. In code, a variant string within a a blockstate JSON is represented by a `ModelResourceLocation`, normally shortened to just "MRL". The default `IStateMapper` uses the block's registry name as the location of the blockstate JSON (block `examplemod:testblock` goes to JSON `examplemod:testblock` goes to filepath `assets/examplemod/blockstates/testblock.json`), and the variant string is pieced together from the blockstate's properties. More information can be found [here][statemapper].
+
+As an example, let's take a look at the vanilla `oak_log.json`:
+
+```json
+{
+  "variants": {
+    "axis=y":  { "model": "oak_log" },
+    "axis=z":   { "model": "oak_log_side" },
+    "axis=x":   { "model": "oak_log_side", "y": 90 },
+    "axis=none":   { "model": "oak_bark" }
+  }
+}
+```
+
+Here we define 4 variant strings, and for each we use a certain model, either the upright log, the sideways log (rotated or not), and the all bark model (this model is not seen normally in vanilla; you have to use `/setblock` to create it). Since logs use the default `IStateMapper`, these variants will define the look of a log depending on the property `axis`.
+
+A blockstate always has to be defined for all possible variant strings. When you have many properties, this results in lots of possible variants, as each and every combination of properties must be defined. In Minecraft 1.8's blockstate format, you have to define each and every string explicitly, which leads to long, complicated files. It also doesn't support the concept of submodels, or multiple models in the same blockstate. In order to allievate this, Forge introduced its [own blockstate format][Forge blockstate], which is available in Minecraft 1.8 and up.
+
+Starting from Minecraft 1.9, Mojang also introduced the "multipart" format. You can find a definition of its format on the [wiki][]. Forge's format and the multipart format are not better than each other; they each cover different use cases and it is your choice which one you want to use. Both formats take care of calculating the set of all possible variants for you behind the scenes, so your files can be simpler.
```

Comment:
don't know if we want to talk about this here but the two systems actually work very differently.

Forge version is just syntax sugar to pregenerate all possible combos.
1.9 multipart bakes every part and then decides *at runtime* which lists of quads to return based on the IBlockState. it's basically a json definition for a smartmodel.